### PR TITLE
reduce regex pattern compilation in Pinot jdbc

### PR DESCRIPTION
**Pull request by sullis**
# Reference

https://www.baeldung.com/java-regex-pre-compile

### CODE REVIEW

**Comment by sullis** [comment]
Ready for review.

File: pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/utils/DriverUtils.java [hunk 1]
```diff
@@ -49,6 +49,8 @@ public class DriverUtils {
   public static final String DRIVER = "pinot";
   public static final Logger LOG = LoggerFactory.getLogger(DriverUtils.class);
   private static final String LIMIT_STATEMENT_REGEX = "\\s(limit)\\s";
+  private static final Pattern LIMIT_STATEMENT_REGEX_PATTERN
```
Comment (Jackie-Jiang) [inline pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/utils/DriverUtils.java:52]: (code style) Can you apply [Pinot Style](https://docs.pinot.apache.org/developers/developers-and-contributors/code-setup#intellij) and reformat the changes

File: pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/utils/DriverUtils.java [hunk 2]
```diff
@@ -212,9 +214,9 @@ public static String getJavaClassName(String columnDataType) {
     return columnsJavaClassName;
   }
 
+
   public static boolean queryContainsLimitStatement(String query) {
-    Pattern pattern = Pattern.compile(LIMIT_STATEMENT_REGEX, Pattern.CASE_INSENSITIVE);
-    Matcher matcher = pattern.matcher(query);
+    Matcher matcher = LIMIT_STATEMENT_REGEX_PATTERN.matcher(query);
     return matcher.find();
```
Comment (Jackie-Jiang) [inline pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/utils/DriverUtils.java:220]: (nit) maybe making it one line

**Review by Jackie-Jiang — APPROVED** [review]

File: pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/utils/DriverUtils.java (hunk 1 above)
Comment (sullis) [reply pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/utils/DriverUtils.java]: Done

File: pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/utils/DriverUtils.java (hunk 2 above)
Comment (sullis) [reply pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/utils/DriverUtils.java]: Done

**Comment by sullis** [comment]
Ready for review.

**Comment by sullis** [comment]
fixed a checkstyle violation.

**Comment by sullis** [comment]
ready to merge.

**Pull request merged**